ipn/ipnlocal/local.go

```diff
@@ -1167,9 +1167,18 @@ func (b *LocalBackend) SetControlClientStatus(c controlclient.Client, st control
 		}
 		b.keyExpired = isExpired
 	}
-
+	lastSuggestedExitNode := b.lastSuggestedExitNode
 	unlock.UnlockEarly()
 
+	if st.NetMap != nil && shouldAutoExitNode() {
+		for _, peer := range st.NetMap.Peers {
+			if peer.StableID() == lastSuggestedExitNode && peer.Online() != nil && !*peer.Online() {
+				b.lastSuggestedExitNode = ""
```

Comment:
Need the lock for that, just move all of this before `unlock.UnlockEarly()` 😄 

Hmm, actually, I think it would be better to do after the lock down on what is currently line 1197 (after `// Lock b again and only do the things that require locking.`) because then it'll line up with when `b.netMap` is updated (this happens inside `b.setNetMapLocked` right before the unlock).